### PR TITLE
Fixed NPE in folder settings of outbox folder

### LINKED ISSUE #5249: NPE when trying to change folder settings of outbox folder
**Describe the bug**
I tried to change my folder settings for outbox folder but app crashes.

**To Reproduce**
1. Tab on sandwich icon to open side menu
2. Tab on 'Manage folders'
3. Tab on 'Outbox' (2nd list item)
4. App crashes

**Expected behavior**
App should show folder settings

**Environment (please complete the following information):**
 - K-9 Mail version: actual main (1a68adc8)
 - Android version: 10

**Logs**
FATAL EXCEPTION: main
    Process: com.fsck.k9.debug, PID: 9972
    java.lang.NullPointerException: cursor.getString(7) must not be null
        at com.fsck.k9.storage.messages.CursorFolderAccessor.getSyncClass(RetrieveFolderOperations.kt:151)
        at com.fsck.k9.mailstore.FolderRepository$getFolderDetails$1.map(FolderRepository.kt:62)
        at com.fsck.k9.mailstore.FolderRepository$getFolderDetails$1.map(FolderRepository.kt:8)
        at com.fsck.k9.storage.messages.RetrieveFolderOperations$getFolder$1.doDbWork(RetrieveFolderOperations.kt:44)
        at com.fsck.k9.mailstore.LockableDatabase.execute(LockableDatabase.java:273)
        at com.fsck.k9.storage.messages.RetrieveFolderOperations.getFolder(RetrieveFolderOperations.kt:32)
        at com.fsck.k9.storage.messages.RetrieveFolderOperations.getFolder(RetrieveFolderOperations.kt:16)
        at com.fsck.k9.storage.messages.K9MessageStore.getFolder(K9MessageStore.kt:94)
        at com.fsck.k9.mailstore.FolderRepository.getFolderDetails(FolderRepository.kt:52)
        at com.fsck.k9.ui.managefolders.FolderSettingsViewModel$loadFolderDetails$2.invokeSuspend(FolderSettingsViewModel.kt:74)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)

### PR DESCRIPTION
**Pull request by plan3d**
Fixes #5249

### CODE REVIEW

File: app/core/src/main/java/com/fsck/k9/mailstore/FolderSettingsProvider.kt [hunk 1]
```diff
@@ -37,8 +38,4 @@ class FolderSettingsProvider(val preferences: Preferences, val account: Account)
 
         editor.commit()
     }
-
-    private fun String?.toFolderClass(defaultValue: FolderClass): FolderClass {
```
Comment (cketti) [inline app/core/src/main/java/com/fsck/k9/mailstore/FolderSettingsProvider.kt:41]: I liked the extension function better. Please revert the changes to this class and copy the extension function to `RetrieveFolderOperations` (I prefer the code duplication over a global extension function on `String?`).

File: app/storage/src/test/java/com/fsck/k9/storage/messages/FolderHelpers.kt [hunk 2]
```diff
@@ -15,7 +15,7 @@ fun SQLiteDatabase.createFolder(
     integrate: Boolean = false,
     inTopGroup: Boolean = false,
     displayClass: String = "NO_CLASS",
-    syncClass: String = "INHERITED",
+    syncClass: String? = "INHERITED",
```
Comment (cketti) [inline app/storage/src/test/java/com/fsck/k9/storage/messages/FolderHelpers.kt:18]: :+1: 

`notifyClass` and `pushClass` could be `null`, too. See https://github.com/k9mail/k-9/blob/783182a718c5a1d4c94e354f6e73941214addfbc/app/core/src/main/java/com/fsck/k9/mailstore/LocalStore.java#L898-L911

File: app/storage/src/test/java/com/fsck/k9/storage/messages/RetrieveFolderOperationsTest.kt [hunk 3]
```diff
@@ -45,6 +45,39 @@ class RetrieveFolderOperationsTest : RobolectricTest() {
         assertThat(result).isTrue()
     }
 
+    @Test
+    fun `get folder without sync class`() {
```
Comment (cketti) [inline app/storage/src/test/java/com/fsck/k9/storage/messages/RetrieveFolderOperationsTest.kt:49]: :+1:

Please make this a test for a typical local folder.

See 
https://github.com/k9mail/k-9/blob/783182a718c5a1d4c94e354f6e73941214addfbc/app/core/src/main/java/com/fsck/k9/mailstore/LocalStore.java#L898-L911

For default values:
https://github.com/k9mail/k-9/blob/783182a718c5a1d4c94e354f6e73941214addfbc/app/storage/src/main/java/com/fsck/k9/storage/StoreSchemaDefinition.java#L84-L102

**Review by cketti — CHANGES_REQUESTED** [review]

File: app/core/src/main/java/com/fsck/k9/mailstore/FolderSettingsProvider.kt (hunk 1 above)
Comment (plan3d) [reply app/core/src/main/java/com/fsck/k9/mailstore/FolderSettingsProvider.kt:41]: Can you please tell me why you prefer code duplication over something like this:

```
package com.fsck.k9.mail

fun String?.toFolderClass(defaultValue: FolderClass): FolderClass {
    return if (this == null) defaultValue else FolderClass.valueOf(this)
}

enum class FolderClass {
    NO_CLASS, INHERITED, FIRST_CLASS, SECOND_CLASS
}
```
and `import com.fsck.k9.mail.toFolderClass` in `FolderSettingsProvider` and `RetrieveFolderOperations`?

File: app/storage/src/test/java/com/fsck/k9/storage/messages/FolderHelpers.kt (hunk 2 above)
Comment (plan3d) [reply app/storage/src/test/java/com/fsck/k9/storage/messages/FolderHelpers.kt:18]: Done

File: app/storage/src/test/java/com/fsck/k9/storage/messages/RetrieveFolderOperationsTest.kt (hunk 3 above)
Comment (plan3d) [reply app/storage/src/test/java/com/fsck/k9/storage/messages/RetrieveFolderOperationsTest.kt]: Done

File: app/core/src/main/java/com/fsck/k9/mailstore/FolderSettingsProvider.kt (hunk 1 above)
Comment (cketti) [reply app/core/src/main/java/com/fsck/k9/mailstore/FolderSettingsProvider.kt:41]: To limit the scope of the extension functions. `Storage` and `MailStore` are two independent stores that just happen to use the same serialization format for `FolderClass`. But this fact is an implementation detail that shouldn't be exposed.

Comment (plan3d) [reply app/core/src/main/java/com/fsck/k9/mailstore/FolderSettingsProvider.kt]: Done

**Review by cketti — APPROVED** [review]

**Comment by cketti** [comment]
Thanks :+1:

**Pull request merged**